### PR TITLE
Auth UI & Utility Components

### PR DESCRIPTION
**Pull request by rehanmohiuddin**
- Added Login UI
- Added Register UI
- Added Form Reusable Component
- Added useFormValidator Hook
- Added Button
- Added Loader
- Added Util

### CODE REVIEW

**Review by spraveenofficial — APPROVED** [review]
Code looks well structured.

File: src/components/SideNav/index.scss [hunk 1]
```diff
@@ -0,0 +1,37 @@
+@use "../../styles/util.scss";
+@import "../../styles/variables.scss";
+
+nav {
+  @extend .flex-column, .flex-start, .gap-1;
+  height: 100vh;
+  background-color: $white;
+  //   border-radius: 0.3rem;
+  box-shadow: 0.2rem 0.2rem 0.2rem 0.2rem $light-gray;
+  // position: absolute;
```
Comment (nerdyspook) [inline src/components/SideNav/index.scss:10]: You can remove these comments from here.

File: src/components/SideNav/index.scss [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    gap: calc($spacing-unit * 2);
+  }
+  .nav-icon {
+    color: $black;
+    font-size: calc($spacing-unit * 1.5);
+  }
+  img {
+    width: calc($spacing-unit * 4);
+    @extend .padding-full;
+    // margin-bottom: calc();
```
Comment (nerdyspook) [inline src/components/SideNav/index.scss:32]: You can remove this comment as well.

**Review by nerdyspook — APPROVED** [review]
Code structure seems good to me.

**Pull request merged**